src/Exception/MultipartUploadException.php

```diff
@@ -55,4 +60,26 @@ public function getState()
     {
         return $this->state;
     }
+
+    /**
+     * Get file path of the transfer
+     *
+     * @return array
+     */
+    public function getFilePaths()
+    {
+        return $this->filePath;
+    }
+
+    /**
+     * Construct file path
+     *
+     * @param CommandInterface $cmd
+     */
+    private function collectPathInfo(CommandInterface $cmd)
+    {
+        if (empty($this->filePath) && isset($cmd['Bucket']) && isset($cmd['Key'])) {
```

Comment:
This is still coupling the generic MultipartUploadException to S3. Can you refactor this as previously suggested?